core/shared/src/main/scala/org/http4s/Entity.scala

```diff
@@ -19,20 +19,45 @@ package org.http4s
 import cats.Monoid
 import cats.syntax.all._
 import cats.~>
+import fs2.Pure
+import fs2.Stream
 
-final case class Entity[+F[_]](body: EntityBody[F], length: Option[Long] = None) {
-  def translate[F1[x] >: F[x], G[_]](fk: F1 ~> G): Entity[G] = Entity(body.translate(fk), length)
+sealed trait Entity[+F[_]] {
```

Comment:
So, IDK when's the right time to start discussing this one (feel free to defer :) but pros/cons about `seal`ing `Entity`?

Reply:
I was vaguely thinking we should seal `Entity` but then give it an unsealed subtrait called something like `Other`, so people can extend that but we still get exhaustive matches on `Entity`. But I think we can defer this decision for now.